File: backend/app/schemas/imports.py
from datetime import datetime

from pydantic import BaseModel, Field


class ImportMappingUpdate(BaseModel):
    user_id: str
    mapping: dict[str, str]


class ImportConfirm(BaseModel):
    user_id: str


class ImportErrorRead(BaseModel):
    row_number: int
    field_name: str | None
    error_code: str
    message: str


class ImportBatchRead(BaseModel):
    id: str
    department_id: str
    project_id: str
    created_by: str
    kind: str
    original_name: str
    sha256: str
    size_bytes: int
    mapping: dict[str, str]
    status: str
    total_rows: int
    success_rows: int
    failed_rows: int
    created_at: datetime
    confirmed_at: datetime | None
    completed_at: datetime | None
    error_message: str | None


class ImportPreview(BaseModel):
    batch: ImportBatchRead
    rows: list[dict[str, str | int | float | None]]
    errors: list[ImportErrorRead]


class ImportUploadResponse(ImportPreview):
    detected_headers: list[str] = Field(default_factory=list)

File: frontend/src/types/imports.ts
export type ImportBatch = {
  id: string;
  department_id: string;
  project_id: string;
  created_by: string;
  kind: "sales" | "inventory";
  original_name: string;
  sha256: string;
  size_bytes: number;
  mapping: Record<string, string>;
  status: string;
  total_rows: number;
  success_rows: number;
  failed_rows: number;
  created_at: string;
  confirmed_at: string | null;
  completed_at: string | null;
  error_message: string | null;
};

export type ImportIssue = {
  row_number: number;
  field_name: string | null;
  error_code: string;
  message: string;
};

export type ImportPreview = {
  batch: ImportBatch;
  rows: Array<Record<string, string | number | null>>;
  errors: ImportIssue[];
  detected_headers?: string[];
};
